cwltool/builder.py

```diff
@@ -215,13 +215,11 @@ def generate_arg(self, binding):  # type: (Dict[Text,Any]) -> List[Text]
 
         l = []  # type: List[Dict[Text,Text]]
         if isinstance(value, list):
-            if binding.get("itemSeparator"):
+            if binding.get("itemSeparator") and value:
                 l = [binding["itemSeparator"].join([self.tostr(v) for v in value])]
             elif binding.get("valueFrom"):
                 value = [self.tostr(v) for v in value]
                 return ([prefix] if prefix else []) + value
-            elif prefix:
-                return [prefix]
```

Comment:
This breaks https://ci.commonwl.org/job/cwltool-pr-conformance-multiver/961/version=v1.0/testReport/py2/7/Test_nested_prefixes_with_arrays/